README.md

```diff
@@ -134,33 +185,51 @@ In `docker-compose.yml`, the following ports on the docker host are connected t
+Because IoT data is generally persistent, we decided that the extra level of indirection was required. To help find things, consult the following table. Data files are kept in the following locations by default.
 
-| **Component** | **Data file location on host**| **Location in container**  |
-|---------------|-----------|----------------------------|
-| Node-RED      | `${IOT_DASHBOARD_DATA}node-red`| /data
-| InfluxDB      |  `${IOT_DASHBOARD_DATA}influxdb` | /var/lib/influxdb
-| Grafana       | `${IOT_DASHBOARD_DATA}grafana` | /var/lib/grafana|
-| Mqtt | `${IOT_DASHBOARD_DATA}mqtt/credentials` | /etc/mosquitto/credentials
-| Nginx | `${IOT_DASHBOARD_DATA}docker-nginx/authdata`| /etc/nginx/authdata
-| Let's Encrypt certificates |`${IOT_DASHBOARD_DATA}docker-nginx/letsencrypt`|/etc/letsencrypt
+| **Component** | **Data file location on host**                | **Location in container**     |
+|---------------|-----------------------------------------------|-------------------------------|
+| Node-RED      | `${IOT_DASHBOARD_DATA}node-red`               | /data                         |
+| InfluxDB      |  `${IOT_DASHBOARD_DATA}influxdb`              | /var/lib/influxdb             |
+| Grafana       | `${IOT_DASHBOARD_DATA}grafana`                | /var/lib/grafana              |
+| Mqtt          | `${IOT_DASHBOARD_DATA}mqtt/credentials`       | /etc/mosquitto/credentials    |
+| Nginx         | `${IOT_DASHBOARD_DATA}docker-nginx/authdata`  | /etc/nginx/authdata           |
+| Let's Encrypt certificates |`${IOT_DASHBOARD_DATA}docker-nginx/letsencrypt`| /etc/letsencrypt |
+| DNCserver     | `${IOT_DASHBOARD_DATA}apiserver/dncserver`    | /apiserver/dncserver          |
+| DNCgiplugin   | `${IOT_DASHBOARD_DATA}apiserver/dncgiplugin`  | /apiserver/dncgiplugin        |
+| DNCstdplugin  | `${IOT_DASHBOARD_DATA}apiserver/dncstdplugin` | /apiserver/dncstdplugin       |
+| Mongodb       | `${IOT_DASHBOARD_DATA}mongodb/mongodb_data`   | /data/db                      |
+| Expo          | `${IOT_DASHBOARD_DATA}expo/dncui`             |  /expo/dncui                  |
 
-As shown, one can easily change locations on the **host** (e.g. for testing). This can be done by setting the environment variable `IOT_DASHBOARD_DATA` to the **absolute path** (with trailing slash) to the containing directory prior to
-calling `docker-compose up`. The above paths are appended to the value of `IOT_DASHBOARD_DATA`. Directories are created as needed.
+As shown, one can easily change locations on the **host** (e.g. for testing). This can be done by setting the environment variable `IOT_DASHBOARD_DATA` to the **absolute path** (with trailing slash) to the containing directory prior to
+calling `docker-compose up`. The above paths are appended to the value of `IOT_DASHBOARD_DATA`. Directories are created as needed.
 
-Normally, this is done by an appropriate setting in the `.env` file.
+Normally, this is done by an appropriate setting in the `.env` file.
```

Comment:
I don't understand what changed here. Did the blank turn into something else (tab or something)?

Reply:
Yes. In visual studio code, it shows as small box(square) around `.env`